### ISSUE TITLE
Force estimator seems not thread-safe

### ISSUE THREAD

**Issue by ye-luo** [QMCPACK/qmcpack#2129]
When debugging #2110, I noticed that a VMC run can crash if OMP_NUM_THREADS is large like 32.

**Comment by jtkrogel** [comment]
Do you have inputs to reproduce the problem?

**Comment by ye-luo** [comment]
#2110 has input files.

**Comment by tiihonej** [comment]
I agree there's a problem. A sample VMC run crashes immediately with the force estimator whenever OMP_NUM_THREADS>1, but works fine without it. 

Sorry for taking to time to check this; been only running MPI so far. This does not seem to be an urgent matter for anyone else right now, but will have to be fixed for later applications of the forces. I'll try and pin down the cause sometime soon in the future.
